azureFunctionsPackage task fails when using Azure Functions Plugin for Gradle 1.15.0

When updating to Azure Functions Plugin for Gradle 1.15.0, `azureFunctionsPackage` task now fails with the error below.
```
Execution failed for task ':azureFunctionsPackage'.
> generate configuration files and prepare staging directory
```

This does not occur before 1.14.0.

Here is a sample repository:
https://github.com/taisuke-fujimoto/azure-gradle-plugins-1.15.0-bug

Checked on my PC (Windows 10) and GitHub Actions (Ubuntu latest).

same for me, running it with --stacktrace I see the following stacktrace:
`
Caused by: org.gradle.api.GradleException: Cannot package functions due to error: null
        at com.microsoft.azure.plugin.functions.gradle.task.PackageTask.build(PackageTask.java:57)
        ... 122 more
Caused by: java.lang.NullPointerException
        at com.microsoft.azure.toolkit.lib.common.operation.OperationContext.setTelemetryProperties(OperationContext.java:46)
        at com.microsoft.azure.plugin.functions.gradle.task.PackageTask.build(PackageTask.java:44)
        ... 122 more

`

--- a/azure-functions-gradle-plugin/gradle.properties
+++ b/azure-functions-gradle-plugin/gradle.properties
@@ -21,7 +21,7 @@
 # SOFTWARE.
 
 group                 = com.microsoft.azure
-version               = 1.16.0
+version               = 1.16.1
 
 pluginId                  = com.microsoft.azure.azurefunctions
 pluginShortName           = azurefunctions

--- a/azure-functions-gradle-plugin/src/main/java/com/microsoft/azure/plugin/functions/gradle/GradleFunctionContext.java
+++ b/azure-functions-gradle-plugin/src/main/java/com/microsoft/azure/plugin/functions/gradle/GradleFunctionContext.java
@@ -206,14 +206,14 @@ public class GradleFunctionContext {
     public Map<String, String> getTelemetryProperties() {
         final Map<String, String> result = new HashMap<>();
         final GradleRuntimeConfig runtime = getRuntime();
-        final String javaVersion = runtime == null ? null : runtime.javaVersion();
-        final String os = runtime == null ? null : runtime.os();
-        final boolean isDockerFunction = runtime != null && StringUtils.isNotEmpty(runtime.image());
-        result.put(FUNCTION_JAVA_VERSION_KEY, StringUtils.isEmpty(javaVersion) ? "" : javaVersion);
-        result.put(FUNCTION_RUNTIME_KEY, StringUtils.isEmpty(os) ? "" : os);
+        final String javaVersion = Optional.ofNullable(runtime).map(GradleRuntimeConfig::javaVersion).orElse(StringUtils.EMPTY);
+        final String os = Optional.ofNullable(runtime).map(GradleRuntimeConfig::os).orElse(StringUtils.EMPTY);
+        final boolean isDockerFunction = Optional.ofNullable(runtime).map(GradleRuntimeConfig::image).filter(StringUtils::isNotBlank).isPresent();
+        result.put(FUNCTION_JAVA_VERSION_KEY, javaVersion);
+        result.put(FUNCTION_RUNTIME_KEY, os);
         result.put(FUNCTION_IS_DOCKER_KEY, String.valueOf(isDockerFunction));
-        result.put(FUNCTION_REGION_KEY, getRegion());
-        result.put(FUNCTION_PRICING_KEY, StringUtils.isEmpty(getPricingTier()) ? "" : getPricingTier());
+        result.put(FUNCTION_REGION_KEY, Optional.ofNullable(getRegion()).orElse(StringUtils.EMPTY));
+        result.put(FUNCTION_PRICING_KEY, Optional.ofNullable(getPricingTier()).orElse(StringUtils.EMPTY));
         result.put(DISABLE_APP_INSIGHTS_KEY, String.valueOf(isDisableAppInsights()));
         result.put(DEPLOY_TO_SLOT_KEY, String.valueOf(StringUtils.isEmpty(getDeploymentSlotName())));
         return result;

--- a/azure-webapp-gradle-plugin/src/main/java/com/microsoft/azure/plugin/webapps/gradle/AzureWebappPluginExtension.java
+++ b/azure-webapp-gradle-plugin/src/main/java/com/microsoft/azure/plugin/webapps/gradle/AzureWebappPluginExtension.java
@@ -250,8 +250,8 @@ public class AzureWebappPluginExtension {
         result.put(OS_KEY, os);
         result.put(JAVA_VERSION_KEY, java.util.Optional.ofNullable(runtime).map(GradleRuntimeConfig::javaVersion).orElse(StringUtils.EMPTY));
         result.put(JAVA_WEB_CONTAINER_KEY, java.util.Optional.ofNullable(runtime).map(GradleRuntimeConfig::webContainer).orElse(StringUtils.EMPTY));
-        result.put(PRICING_TIER_KEY, pricingTier);
-        result.put(REGION_KEY, region);
+        result.put(PRICING_TIER_KEY, java.util.Optional.ofNullable(pricingTier).orElse(StringUtils.EMPTY));
+        result.put(REGION_KEY, java.util.Optional.ofNullable(region).orElse(StringUtils.EMPTY));
         if (runtime != null && StringUtils.equalsIgnoreCase(os, OperatingSystem.DOCKER.getValue())) {
             final boolean isCustomRegistry = StringUtils.isNotEmpty(runtime.registryUrl());
             final DockerImageType imageType;

